students/WWhite/lesson05/Mailroom.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+# -------------------------------------#
+# Desc: Mailroom Part 2
+# Dev: Will White
+# Date: 5/8/2018
+# ChangeLog: (When,Who,What)
+# -------------------------------------#
+
+
+dict_donations = {
+    "bill gates": [25000.00, 3], "monet holt": [50000.00, 2], "jeff bezos": [123500.09, 3],
+    "john wick": [120000.00, 2], "john snow": [10.56, 1]
+}  # List of donors
+
+
+def menu_options():  # Function to run the menu options
+    user_input = input('''
+Please enter a number from the following options:
+
+[1] Send a Thank You
+[2] Create a Report
+[3] Send Letters to Everyone
+[4] Quit the Program
+
+''')
+    return user_input  # Function returns user_input
+
+
+def prompt_for_name():  # Function to prompt the user for a name
+    donor_name = input("Please input the donor's name, or type 'list' to see a list of donor names: \n")
+    while donor_name == 'list':  # If user inputs "list", program will print a list of current donors
+        for key in dict_donations:
+            print(key.title())
+        donor_name = input("\nPlease input the donor's name: \n").lower()
+    return donor_name  # Function returns the donor's name
+
+
+def print_letter(donor_name, donation_amount):
+    dict_donor_info = {'name': donor_name.title(), 'donation': donation_amount}
+    str_letter = '''
+Dear {name},
+
+Thank you for your donation of {donation}, it is very much appreciated.
+
+Kind Regards,
+Your Favorite Local Charity
+'''.format(**dict_donor_info) # Thank you email with formatting to include donor name and amount
+    return str_letter
+
+
+def add_donation(donor_name):  # Function for the user to add a donation
+    while True:
+        try:
+            donation_amount = input("Please input the donation amount: ")
+            donation_amount = float(donation_amount)
+            break
+        except:
```

Comment:
It's a good idea to identify and catch the expected exception in your try/except statements, so would recommend using `except ValueError:`